arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+          ? string.replace(reRegExpChar, '\\$&')
+          : string;
+      };
+    }());
+
+    var getKeysRegExp = function (map) {
+      var keys = Object.keys(map).sort(function (a, b) {
+        return b.length - a.length;
+      });
+
+      return new RegExp(keys.map(escapeRegExp).join('|'), 'gi');
+    };
+
+    var identifiersMap = MathJax.Hub.config.Arabic.identifiersMap;
+    var identifiersKeysRegExp = getKeysRegExp(identifiersMap);
+
+    var operatorsMap = MathJax.Hub.config.Arabic.operatorsMap;
+    var operatorsKeysRegExp = getKeysRegExp(operatorsMap);
+
+
+    TEX.Definitions.Add({
+      macros: {
+        'ar': 'HandleArabic',
+        'alwaysar': 'MarkAsArabic',
+        'fliph': 'HandleFlipHorizontal',
+        'transx': 'TranslateTeX',
+        'transt': 'TranslateText',
+        'transs': 'TranslateSymbols'
+      }
+    });
+
+
+    TEX.Definitions.Add({
+      macros: function () {
+        var definitions = {};
+
+        Object.keys(dict).forEach(function (key) {
+          var texCommand = dict[key][0];
+          definitions[texCommand] = key;
+        });
+
+        return definitions;
+      }()
+    });
+
+
+    TEX.Parse.Augment(function () {
+      var parsers = {};
+
+      Object.keys(dict).forEach(function (key) {
+        parsers[key] = dict[key][1]; // Parser function
+      });
+
+      return parsers;
+    }());
+
+
+    TEX.Parse.Augment({
+      flipHorizontal: function (token) {
+        token.arabicFlipH = !token.arabicFlipH;
+          // Invert the value, because flipping twice means, it is not flipped
+        return token;
+      },
+      arabicNumber: function (token) {
+        var numbersMap = MathJax.Hub.config.Arabic.numbersMap;
+        var text = token.data[0].data[0];
+        var mapped = text.replace(englishNumbersRegExp, function (m) {
+          return numbersMap[m];
+        });
+
+        if (mapped !== text) {
+          token.data[0].data[0] = mapped;
+          token.arabicFontLang = 'ar';
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicIdentifier: function (token) {
+        var text = token.data[0].data[0];
+
+        if ('chars' === token.data[0].type) {
+          // English Symbols like X and Y
+          var mapped = text.replace(identifiersKeysRegExp, function (m) {
+            return identifiersMap[m.toLowerCase()];
+          });
+
+          if (mapped !== text) {
+            token.data[0].data[0] = mapped;
+            token.arabicFontLang = 'ar';
+          }
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicOperator: function (token) {
+        var text = token.data[0].data[0];
+        var mapped = text.replace(operatorsKeysRegExp, function (m) {
+          return operatorsMap[m];
+        });
+
+        if (mapped !== text) {
+          token = this.flipHorizontal(token);
+          token.arabicFontLang = 'ar';
+          token.data[0].data[0] = mapped;
+        }
```

Comment:
Well, essentially yes. The operators are different. For example I don't need map anything to the following operators: `\le`, `\gt`, `\not\subset`, etc since they're the same in Arabic. However, I still need to flip them regardless. For other operators like the comma `,` and the `\lim` I actually do map them, and need to flip them as well.

However, for `arabicIdentifier`, I don't have a substitute all the time, for example `s=1` will display the `s` character in English without flipping it, since I don't have a substitute for it.

So I think the logic is correct so far.